### PR TITLE
feat: add support for creation of cloudgov only projects

### PR DESCRIPTION
**Pull request by gssbzn**
## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/projects.go [hunk 1]
```diff
@@ -63,12 +63,13 @@ var _ ProjectsService = &ProjectsServiceOp{}
...
-	Name         string  `json:"name,omitempty"`
-	ClusterCount int     `json:"clusterCount,omitempty"`
-	Created      string  `json:"created,omitempty"`
-	Links        []*Link `json:"links,omitempty"`
+	ID                      string  `json:"id,omitempty"`
+	OrgID                   string  `json:"orgId,omitempty"`
+	Name                    string  `json:"name,omitempty"`
+	ClusterCount            int     `json:"clusterCount,omitempty"`
+	Created                 string  `json:"created,omitempty"`
+	RegionUsageRestrictions string  `json:"regionUsageRestrictions,omitempty"` // RegionUsageRestrictions for cloud.mongodbgov.com, valid values are GOV_REGIONS_ONLY, COMMERCIAL_FEDRAMP_REGIONS_ONLY, NONE
```
Comment (gssbzn) [inline mongodbatlas/projects.go:71]: just added this field but the reformatting makes it look like a bigger change

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by threebee — APPROVED** [review]

**Review by colm-quinn — APPROVED** [review]
LGTM

**Comment by gssbzn** [comment]
@themantissa you probably want a review here for terraform later on

**Review by themantissa — APPROVED** [review]
Thank you so much for this and for the tag!   Did not know we'll need this for terraform so will add to the fedramp work for it.

**Pull request merged**